server.js

```diff
@@ -133,3 +147,96 @@ async function pinningNode () {
     console.log('peer joined room', topic, peer)
   }
 }
+
+pinningNode()
+
+
+
+const getProfile = async (rootStoreAddress) => {
+  // const profile = getProfile(req.params.address)
+  const rootStore = await orbitdb.open(rootStoreAddress)
+  const readyPromise = new Promise((resolve, reject) => {
+    rootStore.events.on('ready', resolve)
+  })
+  rootStore.load()
+  await readyPromise
+
+  await Promise.resolve(resolve => {
+    rootStore.events.on('replicated', resolve)
+  })
+
+  const profileEntry = rootStore
+    .iterator({ limit: -1 })
+    .collect()
+    .find(entry => {
+      return entry.payload.value.odbAddress.split('.')[1] === 'public'
+    })
+
+    console.log('publicStore')
+    console.log(profileEntry.payload.value.odbAddress)
+    const publicStore = await orbitdb.open(profileEntry.payload.value.odbAddress)
+    const readyPromisePublic = new Promise((resolve, reject) => {
+      publicStore.events.on('ready', resolve)
+    })
+    publicStore.load()
+    await readyPromisePublic
+    await Promise.resolve(resolve => {
+    publicStore.events.on('replicated', resolve)
+  })
+
+    const profile = publicStore.all()
+    console.log(profile)
+
+    await rootStore.close()
+    await publicStore.close()
+
+    // return profile
+    let parsedProfile = {}
+    Object.keys(profile).map(key => { parsedProfile[key] = profile[key].value })
+    // console.timeEnd('get Profile')
+    return parsedProfile
+
+}
+
+const app = express();
+
+app.get("/profile", async (req, res, next) => {
+  const address = req.query.address.toLowerCase()
+  const request = `${ADDRESS_SERVER_URL}/odbAddress/${address}`
+  const getRes = await axios.get(request)
+  const rootStoreAddress = getRes.data.data.rootStoreAddress
+  const cacheProfile = await cache.read(rootStoreAddress)
+  const profile = cacheProfile || await getProfile(rootStoreAddress)
+  res.json(profile)
+  if (!cacheProfile) cache.write(rootStoreAddress, profile)
+});
+
+// TODO return {address: profile} or return array of [{address: profile}].
+app.get("/profileList", async (req, res, next) => {
+  const addressArray = req.query.addressList.split(',').map(val => val.toLowerCase())
+  const request = `${ADDRESS_SERVER_URL}/odbAddresses/`
+  const getRes = await axios.post(request, { identities: addressArray })
+  const rootStoreAddresses = getRes.data.data.rootStoreAddresses
+
+  const profilePromiseArray = Object.keys(rootStoreAddresses)
+    .filter((key) => !!rootStoreAddresses[key])
+    .map(async (key) => {
+      const rootStoreAddress = rootStoreAddresses[key]
+      const cacheProfile = await cache.read(rootStoreAddress)
+      const profile = cacheProfile || await getProfile(rootStoreAddress)
+      if (!cacheProfile) cache.write(rootStoreAddress, profile)
+      return {rootStoreAddress, profile}
+  })
+
+  const profiles = await Promise.all(profilePromiseArray)
+  const parsed = profiles.reduce((acc, val) => {
+    acc[val['rootStoreAddress']] = val['profile']
```

Comment:
you that should be eth address, let me make sure the names are clear as well, that was just recently change to use eth-addr instead of rootstoreaddress. The return obj is   {ethaddress: addressprofileObj, ehtaddress2: ....., ....}

Reply:
Is it really? looks like the `profiles` object is [{rootStoreAddress, profile}, ...], so the return obj would become {rootStoreAddress: profileObj, ...}

Reply:
meant to say request was changed from using root addresses to using eth address, but yes the response is currently incorrect as you described